venear-contract/src/storage.rs

```diff
@@ -0,0 +1,88 @@
+use crate::*;
+use common::near_add;
+use near_sdk::Promise;
+
+#[near(serializers=[json])]
+pub struct StorageBalance {
+    pub total: NearToken,
+    pub available: NearToken,
+}
+
+#[near(serializers=[json])]
+pub struct StorageBalanceBounds {
+    pub min: NearToken,
+    pub max: Option<NearToken>,
+}
+
+impl Contract {
+    fn internal_storage_balance_of(&self, account_id: &AccountId) -> Option<StorageBalance> {
+        if self.accounts.contains_key(account_id) {
+            Some(StorageBalance {
+                total: self.storage_balance_bounds().min,
+                available: NearToken::from_near(0),
+            })
+        } else {
+            None
+        }
+    }
+}
+
+#[near]
+impl Contract {
+    /// Registers a new account. If the account is already registered, it refunds the attached
+    /// deposit.
+    #[payable]
+    pub fn storage_deposit(&mut self, account_id: Option<AccountId>) -> StorageBalance {
+        let amount = env::attached_deposit();
+        let account_id = account_id.unwrap_or_else(env::predecessor_account_id);
```

Comment:
Should this be `env::predecessor_account_id**()**`?

Reply:
It's rust magic. Equivalent to `|| env::predecessor_account_id()`